_posts/2022-12-16-month-of-mlops-third-prize-cheque-easy.md

```diff
@@ -0,0 +1,193 @@
+
+You can also check out a video summary of my submission.
+
+<div class="embed-responsive embed-responsive-16by9 mb-5">
+  <iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/b0_xSISvcfU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
+Based on the requirements of the competition and the usecase I had chosen, I defined the scope of my solution to include the following features:
+
+* Support for labelling data which can be used to train the model
+* Combine the images and GT to prepare a dataset for model training
+* Training a model to extract relevant data from a cheque image
+* Evaluating the performance of the trained model
+* Deploying the model based on a trigger to a REST API endpoint
+* Using trained model for inference on unseen data
+* Creating a demo app for the model 
+
+
+Based on the features I wanted for the solution, it was clear that I needed pipelines for labelling, data processing, model training as well as model inference.
+
+One great thing about using ZenML to setup pipelines is that it makes everything easily reproducible. Each pipeline is made up a sequence of steps and the outputs corresponding to each step are saved in the artifact store corresponding to your ZenML stack - making it very easy to track things.
+
+With ZenML it was also very easy to pick and choose different components to setup stacks (infrastructure) for running the created pipelines. You could even create different stacks to run different kinds of pipelines depending on your use case. I used the stacks described below to build the necessary pipelines for my solution.
+
+Stack for Annotation & Data Processing pipelines included the following components:
+
+* Orchestrator: Local Orchestrator
+* Artifact Storage: Azure Blob storage
+* Secrets Manager: Azure Keyvault
+* Annotator: Label Studio
+
+Stack for Training & Inference pipelines included the following components:
+
+* Orchestrator: Local Orchestrator
+* Artifact Storage: Local Storage
+* Secrets Manager: Local Secrets Manager
+* Experiment Tracker: [MLflow](https://mlflow.org/)
+* Model Deployer: [MLflow](https://mlflow.org/)
+
+
+## Annotation Pipelines
+While I did use a [Kaggle dataset](https://www.kaggle.com/datasets/medali1992/cheque-images) to fine tune my model. I also wanted to keep my solution as close to the real-world setting as possible. In my experience, companies have to spend a good amount of time on annotation when solving complex problems involving use of unstructured data. So I thought it would be useful if the solution could include support for annotating new data to train the model. This component consisted of 2 small pipelines:
```

Comment:
```suggestion
While I did use a [Kaggle dataset](https://www.kaggle.com/datasets/medali1992/cheque-images) to fine tune my model, I also wanted to keep my solution as close to the real-world setting as possible. In my experience, companies have to spend a good amount of time on annotation when solving complex problems involving use of unstructured data. So I thought it would be useful if the solution could include support for annotating new data to train the model. This component consisted of 2 small pipelines:
```